test/new-e2e/tests/windows/install-test/install_test.go

```diff
@@ -110,40 +116,136 @@ func (is *agentMSISuite) TestUpgrade() {
+		is.T().FailNow()
+	}
+
+	// TODO: we shouldn't have to start the agent manually after rollback
+	//       but the kitchen tests did too.
+	err = windows.StartService(t.host, "DatadogAgent")
+	is.Require().NoError(err, "agent service should start after rollback")
+
+	if !previousTester.TestExpectations(is.T()) {
+		is.T().FailNow()
+	}
+
+	previousTester.TestUninstall(is.T(), filepath.Join(outputDir, "uninstall.log"))
+}
+
+// TC-INS-001
+func (is *agentMSISuite) TestRepair() {
+	outputDir, err := runner.GetTestOutputDir(runner.GetProfile(), is.T())
+	is.Require().NoError(err, "should get output dir")
+	is.T().Logf("Output dir: %s", outputDir)
+
+	vm := is.Env().RemoteHost
+	is.prepareHost()
+
+	t := is.installAgent(vm, "", filepath.Join(outputDir, "install.log"))
+
+	err = windows.StopService(t.host, "DatadogAgent")
+	is.Require().NoError(err)
+
+	// Corrupt the install
+	err = t.host.Remove("C:\\Program Files\\Datadog\\Datadog Agent\\bin\\agent.exe")
+	is.Require().NoError(err)
+	err = t.host.RemoveAll("C:\\Program Files\\Datadog\\Datadog Agent\\embedded3")
+	is.Require().NoError(err)
+
+	if !is.Run("repair install", func() {
+		err = windowsAgent.RepairAllAgent(t.host, "", filepath.Join(outputDir, "repair.log"))
+		is.Require().NoError(err)
+	}) {
+		is.T().FailNow()
+	}
+
+	if !t.TestExpectations(is.T()) {
+		is.T().FailNow()
+	}
+
+	t.TestUninstall(is.T(), filepath.Join(outputDir, "uninstall.log"))
+}
+
+// installAgent installs the agent package on the VM and returns the Tester
+func (is *agentMSISuite) installAgent(vm *components.RemoteHost, args string, logfile string, testerOpts ...TesterOption) *Tester {
+	opts := []TesterOption{
+		WithAgentPackage(is.agentPackage),
+	}
+	opts = append(opts, testerOpts...)
+	t, err := NewTester(is.T(), vm, opts...)
+	is.Require().NoError(err, "should create tester")
+
+	if !is.Run(fmt.Sprintf("install %s", t.agentPackage.AgentVersion()), func() {
+		err = t.InstallAgent(is.T(), args, logfile)
+		is.Require().NoError(err, "should install agent %s", t.agentPackage.AgentVersion())
+	}) {
+		is.T().FailNow()
+	}
+
+	return t
+}
+
+// installLastStable installs the last stable agent package on the VM, runs tests, and returns the Tester
+func (is *agentMSISuite) installLastStable(vm *components.RemoteHost, args string, logfile string) *Tester {
+	previousAgentPackage, err := windowsAgent.GetLastStablePackageFromEnv()
+	is.Require().NoError(err, "should get last stable agent package from env")
+
+	t, err := NewTester(is.T(), vm,
+		WithAgentPackage(previousAgentPackage),
+		WithPreviousVersion(),
+	)
+	is.Require().NoError(err, "should create tester")
+
+	if !is.Run(fmt.Sprintf("install %s", previousAgentPackage.AgentVersion()), func() {
+		err = t.InstallAgent(is.T(), args, logfile)
+		is.Require().NoError(err, "should install agent %s", previousAgentPackage.AgentVersion())
+	}) {
+		is.T().FailNow()
+	}
+
+	if !t.TestExpectations(is.T()) {
+		is.T().FailNow()
```

Comment:
I think it is useful to do some simple checks to ensure the previous agent installed and running okay since that is the assumption going into the rest of the test.